backend/model/model.go

```diff
@@ -1590,17 +1590,38 @@ func MigrateDB(ctx context.Context, DB *gorm.DB) (bool, error) {
 	// ignore errors - an error means that there are no partitions, so we can safely use the zero-value.
 	DB.Raw("select split_part(relname, '_', 5) from pg_stat_all_tables where relname like 'error_object_embeddings_partitioned%' order by relid desc limit 1").Scan(&lastCreatedPart)
 
-	// Make sure partitions are created for the next 1k projects
-	for i := lastCreatedPart; i < lastVal+1000; i++ {
-		sql := fmt.Sprintf(`
+	// Make sure partitions are created for the next 1k projects, starting with the next partition needed
+	for i := lastCreatedPart + 1; i < lastVal+1000; i++ {
+		if err := DB.Exec(fmt.Sprintf(`
 			CREATE TABLE IF NOT EXISTS error_object_embeddings_partitioned_%d
 			PARTITION OF error_object_embeddings_partitioned
```

Comment:
I think this will require the same access exclusive lock and create it as a partition. I think `CREATE TABLE LIKE` would be better here https://www.postgresql.org/docs/current/ddl-partitioning.html#:~:text=parent%20table%27s%20definition%3A-,CREATE%20TABLE%20measurement_y2008m02,-(LIKE%20measurement%20INCLUDING